cert/selfsigned_test.go

```diff
@@ -185,3 +224,63 @@ func TestTLSDisableAutofill(t *testing.T) {
 		"TLS Certificate was not marked as outdated when it should be",
 	)
 }
+
+// TestTlsConfig tests to ensure we can generate a TLS Config from
+// a tls cert and tls key.
+func TestTlsConfig(t *testing.T) {
+	tempDir := t.TempDir()
+	certPath := tempDir + "/tls.cert"
+	keyPath := tempDir + "/tls.key"
+
+	// Generate TLS files with an extra IP and domain.
+	certBytes, keyBytes, err := cert.GenCertPair(
+		"lnd autogenerated cert", certPath, keyPath,
+		[]string{extraIPs[0]}, []string{extraDomains[0]}, false,
+		testTLSCertDuration,
+	)
+	require.NoError(t, err)
+
+	err = cert.WriteCertPair(certPath, keyPath, certBytes, keyBytes)
+	require.NoError(t, err)
+
+	certBytes, err = ioutil.ReadFile(certPath)
+	require.NoError(t, err)
+
+	keyBytes, err = ioutil.ReadFile(keyPath)
+	require.NoError(t, err)
+
+	// Load the certificate.
+	certData, parsedCert, err := cert.LoadCertFromBytes(
+		certBytes, keyBytes,
+	)
+	require.NoError(t, err)
+
+	// Check to make sure the IP and domain are in the cert.
+	var foundDomain bool
+	var foundIp bool
+	for _, domain := range parsedCert.DNSNames {
```

Comment:
nit: use `require.Contains(t, parsedCert.DNSNames, extraDomains[0])`? Same below for the IP addresses.

Reply:
Sounds good. Just a note for the IP addresses ~ this didn't quite work. Had to loop through each of parsedCert.IPAddresses and turn it to a string to make sure the two equate correctly